### ISSUE TITLE
Clarify lookups within complex variables

### ISSUE THREAD

**Issue by czender** [diwg/cf2#6]
In EUMETSAT data, sometimes different channels of the same instrument will have different horizontal resolutions. This implies a different grid per channel for a given swath.

EUMETSAT implemented this as follows:
```
[data group, contains coordinate system with grid_mapping "geostationary", referencing coordinates "y x"
[channel_specific group, defining y and x variables]
[measured group, contains effective radiances] [quality] [etc.]

```
So in the group measured, the geostationary grid mapping is referenced, which is two groups higher in data. This references coordinates y and x. Because the coordinates for the same swath are different for different channels, the actual coordinates referred to are in channel_specific.

This works great, as long as the implementation is such that the coordinate system is looked up in order to obtain the variable names to be looked up in the future, and these variables are looked up using the same scope as is visible from the variable for which the lookup was invoked.

Should be discussed and described.

**Comment by erget** [comment]
When referencing a complex type, such as e.g. a grid mapping, it is possible that the grid mapping contains references to other variables, e.g. via the `coordinates` attribute. In this case, the scoping rules described in the proposal apply to every lookup as if it were the first lookup, i.e. ancillary variables such as `coordinates` are not looked up using the scope visible from the `grid_mapping` variable, but rather that of the original caller (the variable making use of the `coordinates` via the `grid_mapping`).

One use case for this is when a single netCDF file is used to store the observations from all channels of a satellite instrument with variable horizontal resolution. In this case, the `grid_mapping` variable might be shared across several channels, but different channels might make use of different `coordinates` to accommodate their differing viewing characteristics.

A reduced example from MTG looks like this:
```
group:data{
  dimensions:
    index = configured_value;
  variables:
    int mtg_geos_projection;
      mtg_geos_projection:long_name =”MTG geostationary projection”;
      mtg_geos_projection: grid_mapping_name =”geostationary”;
      mtg_geos_projection:perspective_point_height = “35786400”; //m – WGS84 configured value
      mtg_geos_projection:semi_major_axis = “6378137” ;  //m– WGS84 configured value
      mtg_geos_projection:semi_minor_axis = “6356752” ; //m – WGS84 configured value
      mtg_geos_projection:inverse_flattening = ”298.257223563; // WGS84 configured value
      mtg_geos_projection:latitude_of_projection_origin =”0”; // configured value
      mtg_geos_projection:longitude_of_projection_origin =”0”; //  configured value
      mtg_geos_projection:sweep_angle_axis = “x”;
      mtg_geos_projection:units = “m”;
      mtg_geos_projection:coordinates = “y x”;

  // Repeat such groups for each channel in instrument
  group:vis_04{
    group:measured{
      variables:
        short x(x);   // x coordinate variable
          x:long_name = “x coordinate in mtg_geos_projection”;
          x:standard_name = “projection_x_coordinate”
          x:unit  = “radian”;
          x:axis = “X”;
          x:valid_range = 1, 5568;
          x:scale_factor = 5.588777728e-05; // configured value for 2 km grid
          x:add_offset = 0.1555636281; // configured value for 2 km grid

        short y(y);   // y coordinate variable
          y:long_name = “y coordinate mtg_geos_projection”;
          y:standard_name = “projection_y_coordinate”
          y:unit  = “radian”;
          y:axis = “Y”;
          y:valid_range = 1, 5568;
          y:scale_factor = 5.588777728e-05; // configured value for 2 km grid
          y:add_offset = -0.1555636281; // configured value for 2 km grid

        ushort effective_radiance(x,y);
          effective_radiance:long_name = “Effective radiance”;
          effective_radiance:standard_name = “effective_radiance_in_wavenumber”
          effective_radiance:units = “mW.m-2.sr-1.(cm-1)-1”;
          effective_radiance:coordinates = “x y”;
          effective_radiance:grid_mapping = “mtg_geos_projection”;
    } // measured
  } // vis04

  // Other channels. All share mtg_geos_projection but define their own x and y.
  group:ir_105{
    group:measured{
      // ...
    } // ir_105
  } // measured
} // data
```

**Comment by czender** [comment]
Hmm. This seems like a plausible use-case, that could be used to clarify how to implement the existing convention. I think the draft convention can handle this situation by:
1. Eliminate "coordinates" attribute from map_projection. It makes little sense to me to add "coordinates" to a dummy metadata variable like map_projection. As it makes things ambiguous, perhaps the draft should be changed to explicitly recommend against the practice.
2. Physical variables that do need "coordinates" attributes include, e.g., `effective_radiance`. The existing coordinates attribute will find the appropriate x and y as they are in the local group. Furthermore, the grid_mapping attribute attached to `effective_radiance` will find the appropriate ancestor grid_mapping using the existing algorithm.
In other words, I think this use case highlights a new recommendation the proposal should make. Would you care to draft that?

**Comment by erget** [comment]
Just realized I'd never responded to that. I think this is a good way forward. I'll draft something appropriate.
